### PR TITLE
[2201.9.x] Fix creating redundant future values for default value function calls

### LINKED ISSUE #42566: High error rate in load tests for update 9
**Description:**
$Subect.  Seems like this happened right after we bumped the versions for update9

Ref:
https://github.com/ballerina-platform/module-ballerina-http/blob/master/load-tests/h1_h1_passthrough/results/summary.csv#L465

**Steps to reproduce:**

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by HindujaB**
## Purpose
$subject
Fixes #42566 

## Approach
Happens only when record values with defaultable fields are created outside of Ballerina runtime - E.g. : netty environment

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by HindujaB** [comment]
Before the fix:
Screenshot 2024-04-17 at 18 54 38

After the fix:
Screenshot 2024-04-17 at 18 54 55

**Review by hasithaa — APPROVED** [review]

**Review by gabilang — APPROVED** [review]
LGtM

**Review by TharmiganK — APPROVED** [review]
Tested this change with HTTP and now load tests are having 0% error rate:
https://github.com/ballerina-platform/module-ballerina-http/pull/1948/files

Also tested the changes locally by monitoring the memory. There is no high memory usuage, the memory usage is similar to update 8

**Comment by HindujaB** [comment]
The master PR is failing for the Full Build Pipeline. I am checking it currently.

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/ValueUtils.java [hunk 1]
```diff
@@ -232,27 +223,9 @@ public void notifyFailure(BError error) {
...
-                callback.notifySuccess(result);
-            }
-
-            @Override
-            public void notifyFailure(BError error) {
-                callback.notifyFailure(error);
-            }
-        };
-        scheduler.scheduleFunction(new Object[1], func, null, returnType, strandName, metadata, childCallback);
+        parent.scheduler.schedule(new Object[1], func.getFunction(),future);
```
Comment (gabilang) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/ValueUtils.java:228]:
```suggestion
        parent.scheduler.schedule(new Object[1], func.getFunction(), future);
```

**Review by gabilang — APPROVED** [review]

**Pull request merged**